libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -1,18 +1,37 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
+import { addToReadingList, getReadingList, removeFromReadingList } from '@tmo/books/data-access';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { ReadingListItem } from '@tmo/shared/models';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'tmo-reading-list',
   templateUrl: './reading-list.component.html',
   styleUrls: ['./reading-list.component.scss']
 })
-export class ReadingListComponent {
+export class ReadingListComponent implements OnDestroy {
   readingList$ = this.store.select(getReadingList);
+  public snackBarSubscription$: Subscription;
 
-  constructor(private readonly store: Store) {}
+  constructor(
+    private readonly store: Store,
+    private snackBar: MatSnackBar
+  ) {}
 
-  removeFromReadingList(item) {
+  removeFromReadingList(item: ReadingListItem) {
     this.store.dispatch(removeFromReadingList({ item }));
+    this.openSnackBar(item);
```

Comment:
So, here, you are opening the snackbar simultaneously with dispatching the `removeFromReadingList ` action. What happens if the `removeFromReadingList ` is not confirmed, and it fails? Then the snackbar would allow the user to "undo" an action that did not succeed in the first place. You should make sure that `removeFromReadingList ` is confirmed before allowing the user to "undo" it.